### ISSUE TITLE
Changing to custom titlebar while maximized on linux causes rendering issues and double titlebar

### ISSUE THREAD

**Issue by umnovI** [microsoft/vscode#63022]
Issue Type: <b>Bug</b>

If select "window.titleBarStyle" as "custom", as recommend VS Code, then some time it cannot render window if you resize it.
![ezgif-5-cca1abb28faa](https://user-images.githubusercontent.com/30232564/48413535-12e67980-e759-11e8-825f-f4862e94785f.gif)
 

VS Code version: Code 1.29.0 (5f24c93878bd4bc645a4a17c620e2487b11005f9, 2018-11-12T07:42:27.562Z)
OS version: Linux x64 4.4.0-138-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7400 CPU @ 3.00GHz (4 x 3311)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 1, 1|
|Memory (System)|7.70GB (0.20GB free)|
|Process Argv|--unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-apache|mrm|1.1.1
vscode-language-pack-ru|MS-|1.29.2
sublime-keybindings|ms-|4.0.0
vscode-icons|rob|7.28.0
code-settings-sync|Sha|3.2.0
markdown-all-in-one|yzh|1.7.0

</details>

**Comment by sbatten** [comment]
@joaomoreno as a long term user of the custom menus on Linux have you seen this behavior?

**Comment by sbatten** [comment]
@umnovI I'm noticing something strange from your screen shot. You appear to have 2 title bars, both the native and custom. Can you provide info on your linux distro and version. Looks like Ubuntu MATE, which version?

**Comment by umnovI** [comment]
@sbatten 
```
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial

```
`MATE Desktop Environment 1.12.1`

**Comment by sbatten** [comment]
@umnovI hmm, tried with that OS and can't seem to repro these issues. Do they persist after restarting VS Code? Can you share details on your screen resolution and scaling settings?

**Comment by umnovI** [comment]
@sbatten sorry, but scale settings of what? Screen or VS Code?

**Comment by umnovI** [comment]
>Do they persist after restarting VS Code?

@sbatten  Not all the time. Sometimes it works correctly and there's only custom title bar. And window resize works properly.

**Comment by sbatten** [comment]
@umnovI scaling settings of your OS and screen resolution

>  Not all the time. Sometimes it works correctly and there's only custom title bar. And window resize works properly.

but it comes back? after restarting VS Code again?

**Comment by umnovI** [comment]
@sbatten resolution: 1920x1080. I only found fonts scale settings and it's 96

>but it comes back? after restarting VS Code again?

Yes.

**Comment by joaomoreno** [comment]
Never seen a window border when using custom title bars. Looks like an Electron issue :thinking:

**Comment by umnovI** [comment]
I was able to reproduce this behavior on another PC with same OS configuration.

**Comment by sbatten** [comment]
@umnovI woot! I've got a repro, though after restarting VS Code it goes away I can see both of your symptoms the first time around. Will continue investigating.

**Comment by sbatten** [comment]
@umnovI there was an issue that was causing a cached version of the menu to be triggered which was resolved separately. after fixing that, I am no longer able to reproduce this issue on Ubuntu MATE with the steps above. can you confirm?

**Comment by umnovI** [comment]
@sbatten do I need to install VS Code Insiders?
I can check this on Monday. I don't have this machine right now.

**Comment by sbatten** [comment]
@umnovI any luck with this?

**Comment by umnovI** [comment]
@sbatten still can repro. It's not that often tho.
![screenshot at 2018-12-04 10 49 53](https://user-images.githubusercontent.com/30232564/49426734-768d2100-f7b2-11e8-828a-cb220f22c2d8.png)

Code version: 
```
Version: 1.30.0-insider
Commit: 07202fbfb82bdada89472dd4b1c3c699b0eab0b4
Date: 2018-12-04T06:15:17.387Z
Electron: 2.0.12
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
OS: Linux x64 4.4.0-140-generic
```

**Comment by sbatten** [comment]
@umnovI I'm not exactly sure yet *why* this is happening, but so I can understand the severity for you, how often does this occur and does it continue to happen after you have had the custom title bar enabled for some time. in other words, if you change the setting, restart vs code (get into a good state), does it remain in a good state or go back and forth?

**Comment by umnovI** [comment]
@sbatten I cannot say how often this happens for sure. In case with Insiders I didn't change settings because custom title bar by default. I just restarted it a several times and maximized/minimized it.
So it continue to happen after I have had the custom title bar enabled for some time.

I'll use Insiders today as my main editor, so I could say more specifically.

**Comment by umnovI** [comment]
@sbatten If code start as maximized and then you unmaximize it, then bug happens.

**Comment by umnovI** [comment]
@sbatten  After I updated VS Code stable to 1.30.0 I didn't have this behavior, but after restarting PC it appears again.Commit c6e592b2b5770e40a98cb9c2715a8ef89aec3d74

**Comment by umnovI** [comment]
@sbatten 
![screenshot_1](https://user-images.githubusercontent.com/30232564/50150296-8fc3c080-02ce-11e9-96ca-84f52c669830.jpg)



```
Version: 1.31.0-insider
Commit: d85a16f2e4354f2850b4171f84fe0c94b0cea338
Date: 2018-12-18T06:14:48.441Z
Electron: 3.0.10
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Linux x64 4.4.0-140-generic
```

**Comment by sbatten** [comment]
as the custom title bar is no longer going to remain the default, this will be de-prioritized

**Comment by jrupac** [comment]
FWIW, I stumbled upon this issue after noticing it happening in Google Chrome as well when resizing from a maximized window.

I am using MATE 1.20.1. So I think this issue is larger than just VSCode (I was also able to replicate this in VSCode too).

In Chome, the workaround is to disable `Settings > "Use system title bar and borders"` and then re-enable.

**Comment by sbatten** [comment]
@jrupac I appreciate the information. Is there a bug in chrome or chromium tracking this that you could link?

**Comment by jrupac** [comment]
I looked for a while and couldn't find an existing Chromium bug for this. However, I did come across a few other instances of the same issue (e.g., with Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1482716). 

Eventually, I stumbled upon a bug in Marco (the default window manager for MATE): https://github.com/mate-desktop/marco/issues/428. That bug pointed out that removing the (unnecessary for non-netbook users?) package `mate-netbook` seems to address the issue. The actual underlying issue is with `mate-maximus` which is a utility that makes all X windows maximized. See https://github.com/mate-desktop/mate-netbook/issues/14 for that bug.

I removed and purged `mate-netbook` and rebooted and it fixed the issue for me! Confirmed working for VSCode, Chrome, and Firefox.
